qiskit/circuit/library/standard_gates/u.py

```diff
@@ -273,33 +273,25 @@ def __array__(self, dtype=None):
 
     @property
     def params(self):
-        """Get parameters from base_gate.
-
-        Returns:
-            list: List of gate parameters.
-
-        Raises:
-            CircuitError: Controlled gate does not define a base gate
-        """
-        if self.base_gate:
-            # CU has one additional parameter to the U base gate
-            return self.base_gate.params + self._params
-        else:
-            raise CircuitError("Controlled gate does not define base gate for extracting params")
+        """return CU params."""
+        return self._params
 
     @params.setter
     def params(self, parameters):
-        """Set base gate parameters.
-
-        Args:
-            parameters (list): The list of parameters to set.
-
-        Raises:
-            CircuitError: If controlled gate does not define a base gate.
-        """
-        # CU has one additional parameter to the U base gate
-        self._params = [parameters[-1]]
+        """Set CUGate params [theta,phi,lam,gamma] and base_gate(UGate) params [theta,phi,lam]"""
+        self._params = []
+        for single_param in parameters:
+            if isinstance(single_param, ParameterExpression):
+                self._params.append(single_param)
+            else:
+                self._params.append(self.validate_parameter(single_param))
```

Comment:
The` if / else`  logic came from the setter `Instruction.params`; the method that was being inherited.  I saw logic with and without the if / else and chose the one that seemed the lowest risk.  Is it useful to keep here for the instantiation process of a `Gate`?  I know validation happens during the `bind_parameters` and `assign_parameters` process but I did not see any validation during instantiation.

I like the idea of adding the U4Gate since it would seem to complete the set.